File: app/src/test/java/io/intrepid/contest/utils/RxUtilsTest.java
package io.intrepid.contest.utils;

import org.junit.Test;

import io.reactivex.disposables.Disposable;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class RxUtilsTest {

    @Test
    public void unsubscribeDisposable() throws Exception {
        Disposable mockDisposable = mock(Disposable.class);
        RxUtils.unsubscribeDisposable(mockDisposable);
        verify(mockDisposable).dispose();

        // just ensures that there's not exception
        RxUtils.unsubscribeDisposable(null);
    }
}

File: app/src/test/java/io/intrepid/contest/utils/RxUtilsTest.kt
package io.intrepid.contest.utils

import io.reactivex.disposables.Disposable
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

class RxUtilsTest {

    @Test
    @Throws(Exception::class)
    fun unsubscribeDisposable() {
        val mockDisposable = mock(Disposable::class.java)
        RxUtils.unsubscribeDisposable(mockDisposable)
        verify(mockDisposable).dispose()

        // just ensures that there's not exception
        RxUtils.unsubscribeDisposable(null)
    }
}
